src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -427,56 +432,76 @@ def run_consent_request(
         policy: Policy,
         privacy_request: PrivacyRequest,
         identity_data: Dict[str, Any],
-        consent_preferences: List[Consent],
     ) -> bool:
         """Execute a consent request. Return whether the consent request to the third party succeeded.
-        Return True if 200 OK
+        Should only propagate either the entire set of opt in or opt out requests.
+        Return True if 200 OK. Raises a SkippingConsentPropagation exception if no action is taken
+        against the service.
```

Comment:
The bulk of the code in this SaaSConnector.run_consent_request is largely unchanged - there's one degree less of indentation because I don't loop through all the preferences, they are collapsed into a single optional boolean ahead of time as to whether or not we will fire the request and which set of opt-in/opt-out preferences we will propagate.